cmake-ide.el

```diff
@@ -619,15 +620,9 @@ the object file's name just above."
 
 
 (defun cmake-ide--get-project-key ()
-  "Return the directory name to run CMake in, it is the Project Key to store this directory in the hash map."
-  (let ((build-parent-directory (or cmake-ide-build-pool-dir temporary-file-directory))
-        build-directory-name)
-    (setq build-directory-name
-          (if cmake-ide-build-pool-use-persistent-naming
-              (replace-regexp-in-string "[-/= ]" "_" (concat (expand-file-name (cmake-ide--locate-project-dir)) cmake-ide-cmake-opts))
-            (make-temp-name "cmake")))
-    (let ((build-dir (expand-file-name build-directory-name build-parent-directory)))
-      (file-name-as-directory build-dir))))
+  "Return the Project Key to store this directory in the hash map.  It is build from the concatenation of project-dir and cmake-opts."
```

Comment:
You were the one who wrote this originally and I don't know of anyone else who needs this functionality so I'll trust you here.

Reply:
My idea was to have a unique key corresponding to the project-dir + cmake opts, and a unique build dir associated with it in a map.